frontend/apps/reader/modules/readerlink.lua

```diff
@@ -687,137 +820,30 @@ function ReaderLink:onGotoLink(link, neglect_current_location, allow_footnote_po
-            text = _("Add to Wallabag"),
-            callback = function()
-                UIManager:close(dialog)
-                self.ui:handleEvent(Event:new("AddWallabagArticle", link_url))
-            end,
-        })
-    end
-    if Device:canOpenLink() then
-        table.insert(alt_handlers_buttons, {
-            text = _("Open in browser"),
-            callback = function()
-                UIManager:close(dialog)
-                Device:openLink(link_url)
-            end,
-        })
-    end
-
-    -- Check if it is a wikipedia link
-    local wiki_lang, wiki_page = link_url:match([[https?://([^%.]+).wikipedia.org/wiki/([^/]+)]])
-    if wiki_lang and wiki_page then
-        logger.dbg("Wikipedia link:", wiki_lang, wiki_page)
-        -- Ask for user confirmation before launching lookup (on a
-        -- wikipedia page saved as epub, full of wikipedia links, it's
-        -- too easy to click on links when wanting to change page...)
-        -- But first check if this wikipedia article has been saved as EPUB
-        local epub_filename = wiki_page .. "."..string.upper(wiki_lang)..".epub"
-        local epub_fullpath
-        -- either in current book directory
-        local last_file = G_reader_settings:readSetting("lastfile")
-        if last_file then
-            local current_book_dir = last_file:match("(.*)/")
-            local safe_filename = util.getSafeFilename(epub_filename, current_book_dir):gsub("_", " ")
-            local epub_path = current_book_dir .. "/" .. safe_filename
-            if util.pathExists(epub_path) then
-                epub_fullpath = epub_path
-            end
-        end
-        -- or in wikipedia save directory
-        if not epub_fullpath then
-            local dir = G_reader_settings:readSetting("wikipedia_save_dir")
-            if not dir then dir = G_reader_settings:readSetting("home_dir") end
-            if not dir then dir = require("apps/filemanager/filemanagerutil").getDefaultDir() end
-            if dir then
-                local safe_filename = util.getSafeFilename(epub_filename, dir):gsub("_", " ")
-                local epub_path = dir .. "/" .. safe_filename
-                if util.pathExists(epub_path) then
-                    epub_fullpath = epub_path
+    local default_title =  T(_("External link:\n\n%1"), BD.url(link_url))
+    local title = default_title
+
+    for _, fn_button in pairs(self._external_buttons) do
+        local button = fn_button(self, link_url)
+        if button.show_in_dialog_func then
+            local show, button_title = button.show_in_dialog_func()
+            if show and button_title then
+                if title == default_title then
+                    -- The default title is replaced by the first non-default button title.
+                    title = button_title
+                else
+                    -- Every other button title value is appended to the title.
+                    title = title .. "\n\n" .. button_title
```

Comment:
(We'll see if this can happen, and how it works in practice, we might need to find wording that look fine together.)

Reply:
In practice, the titles are appended when there's a wiki page lookup for an existing wiki page. No other usage that I know of.